src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -561,6 +579,39 @@ def _build_pex_description(request: PexRequest) -> str:
     return f"Building {request.output_filename} {desc_suffix}"
 
 
+def _strip_comments_from_pex_json_lockfile(lockfile_bytes: bytes) -> bytes:
+    """Pex does not like the header Pants adds to lockfiles, as it violates JSON.
+
+    Note that we only strip lines starting with `//`, which is all that Pants will ever add. If
+    users add their own comments, things will fail.
+    """
+    return b"\n".join(
+        line for line in lockfile_bytes.splitlines() if not line.lstrip().startswith(b"//")
+    )
+
+
+def _is_probably_pex_json_lockfile(lockfile_bytes: bytes) -> bool:
+    for line in lockfile_bytes.splitlines():
+        if line and not line.startswith(b"//"):
+            # Note that pip/Pex complain if a requirements.txt style starts with `{`.
+            return line.lstrip().startswith(b"{")
+    return False
+
+
+def _pex_lockfile_requirement_count(lockfile_bytes: bytes) -> int:
```

Comment:
@thejcannon proposed instead counting the number of `project_name` instances. That would give us a much more accurate number. The only risk is if Pex ever stopped using that specific key. @jsirois I know you have cautioned about the instability of the lock format - any guidance?

Reply:
I think I already mentioned the APIs were all there and fast in Pex and the right approach was probably to implement a `pex3 lock` subcommand to query.